central/complianceoperator/v2/checkresults/datastore/datastore_impl.go

```diff
@@ -52,3 +73,124 @@ func (d *datastoreImpl) SearchCheckResults(ctx context.Context, query *v1.Query)
 
 	return d.store.GetByQuery(ctx, query)
 }
+
+// CheckResultStats retrieves the scan results stats specified by query
+func (d *datastoreImpl) CheckResultStats(ctx context.Context, query *v1.Query) ([]*ResourceCountByResultByCluster, error) {
+	if ok, err := complianceOperatorSAC.ReadAllowed(ctx); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, sac.ErrResourceAccessDenied
+	}
+
+	cloned := query.Clone()
+	cloned.Selects = []*v1.QuerySelect{
+		search.NewQuerySelect(search.ClusterID).Proto(),
+		search.NewQuerySelect(search.Cluster).Proto(),
+		search.NewQuerySelect(search.ComplianceOperatorScanConfig).Proto(),
+		search.NewQuerySelect(search.ComplianceOperatorScanName).Proto(),
+	}
+	cloned.GroupBy = &v1.QueryGroupBy{
+		Fields: []string{
+			search.ClusterID.String(),
+			search.Cluster.String(),
```

Comment:
Why did you choose to group by both `clusterID` and `cluster`? Is it to avoid issues because of decommissioned cluster name reuse?

Reply:
@md2119 Had to put them in the group by because I put it in the select.  I did that for convenience so I wouldn't have to make a call to cluster store to get the name.